README.md

```diff
@@ -93,72 +73,115 @@ npm install
 
 ## Usage
 
-### Generating a new key pair
+### Installing Support for Key Types
 
-Ed25519:
+In order to use this library, you will need to import and install driver
+libraries for key types you'll be working with via the `use()` method.
 
-```js
-const {Ed25519KeyPair} = require('crypto-ld');
+For example, to use this library with only the `ed25519` key type:
 
-const keyPair = await Ed25519KeyPair.generate();
 ```
+const {cryptoLd} = require('crypto-ld');
+cryptoLd.use(require('ed25519-key-pair');
 
-RSA:
+// With only one key type installed, you do not need to specify key type for
+// most operations
+const keyPair = await cryptoLd.generate(); // generates an ed25519 key pair
+```
 
-```js
-const {RSAKeyPair} = require('crypto-ld');
+To use the library with all supported key types:
 
-const keyPair = await RSAKeyPair.generate();
 ```
+const {cryptoLd} = require('crypto-ld');
+cryptoLd.use(require('ed25519-key-pair')); // ed25519 type
+cryptoLd.use(require('rsa-key-pair')); // rsa type
+cryptoLd.use(require('secp256k1-key-pair')); // secp256k1 type
+cryptoLd.use(require('x25519-key-pair')); // x25519 type
+
+// When using multiple key types, you'll need to specify type when generating
+const edKeyPair = await cryptoLd.generate({type: 'ed25519'});
+const rsaKeyPair = await cryptoLd.generate({type: 'rsa'});
+```
+
+### Generating a new public/private key pair
+
+To generate a new public/private key pair: `cryptoLd.generate(options)`:
 
-### Exporting a public/private key pair
+* `{string} [type]` Optional if only one key type is installed, required otherwise. 
+* `{string} [controller]` Optional controller URI or DID to initialize the
+  generated key. (This will also init the key id.) 
+* `{string} [seed]` Optional deterministic seed value (only supported by some
+  key types, such as `ed25519`) from which to generate the key.
+
+### Importing a key pair from storage
+
+To create an instance of a public/private key pair from data imported from
+storage, use `cryptoLd.from()`:
 
 ```js
-const edKeyPair = await Ed25519KeyPair.generate();
-edKeyPair.id = 'did:ex:123#' + edKeyPair.fingerprint();
+const serializedKeyPair = await loadFromSomeStorage();
```

Comment:
We can just put some example JSON instead of what a serialized key pair looks like.

Reply:
Agreed, I'll change that.